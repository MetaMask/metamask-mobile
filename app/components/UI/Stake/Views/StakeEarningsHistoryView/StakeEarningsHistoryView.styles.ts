import type { Theme } from '../../../../../util/theme/models';
import { StyleSheet } from 'react-native';

const stylesSheet = (params: { theme: Theme }) => {
  const { theme } = params;
  const { colors } = theme;

  return StyleSheet.create({
    mainContainer: {
      flexGrow: 1,
      paddingTop: 8,
      paddingHorizontal: 16,
      backgroundColor: colors.background.default,
      justifyContent: 'space-between',
    },
  });
};

export default stylesSheet;
