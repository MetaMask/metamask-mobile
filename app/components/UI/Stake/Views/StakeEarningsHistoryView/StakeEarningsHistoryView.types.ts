import { TokenI } from '../../../Tokens/types';

export interface StakeEarningsHistoryViewRouteParams {
  key: string;
  name: string;
  params: {
    asset: TokenI;
  };
}
