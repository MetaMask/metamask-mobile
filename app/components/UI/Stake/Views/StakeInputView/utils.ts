import {
  STAKE_INPUT_ACTION_TO_LABEL_MAP,
  STAKE_INPUT_VIEW_ACTIONS,
} from './StakeInputView.types';

export const getStakeInputViewTitle = (
  action: STAKE_INPUT_VIEW_ACTIONS,
  tokenSymbol: string,
  isEth = false,
) => {
  const prefix = STAKE_INPUT_ACTION_TO_LABEL_MAP[action];
  const suffix = isEth ? 'ETH' : tokenSymbol;

  return `${prefix} ${suffix}`;
};
