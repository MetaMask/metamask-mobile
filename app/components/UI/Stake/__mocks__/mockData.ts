import {
  ChainId,
  StakingApiService,
  StakingType,
  type PooledStakes,
  type VaultData,
} from '@metamask/stake-sdk';
import { TokenI } from '../../Tokens/types';
import { Contract } from 'ethers';
import { Stake } from '../sdk/stakeSdkProvider';

export const MOCK_STAKED_ETH_ASSET = {
  balance: '4.9999 ETH',
  balanceFiat: '$13,292.20',
  name: 'Staked Ethereum',
  symbol: 'ETH',
} as TokenI;

export const MOCK_GET_POOLED_STAKES_API_RESPONSE: PooledStakes = {
  accounts: [
    {
      account: '0x0123456789abcdef0123456789abcdef01234567',
      lifetimeRewards: '43927049303048',
      assets: '5791332670714232000',
      exitRequests: [
        {
          // Unstaking
          positionTicket: '2153260738145148336740',
          timestamp: '1727110415000',
          totalShares: '989278156820374',
          withdrawalTimestamp: null,
          exitQueueIndex: '-1',
          claimedAssets: null,
          leftShares: null,
        },
        // Requests below are claimable.
        {
          positionTicket: '515964521392314631201',
          timestamp: '1720539311000',
          totalShares: '99473618267007',
          withdrawalTimestamp: '0',
          exitQueueIndex: '57',
          claimedAssets: '100006626507361',
          leftShares: '0',
        },
        {
          positionTicket: '515964620865932898208',
          timestamp: '1720541495000',
          totalShares: '99473618267007',
          withdrawalTimestamp: '0',
          exitQueueIndex: '57',
          claimedAssets: '100006626507361',
          leftShares: '0',
        },
        {
          positionTicket: '516604671289934191921',
          timestamp: '1720607327000',
          totalShares: '1929478758729790',
          withdrawalTimestamp: '0',
          exitQueueIndex: '58',
          claimedAssets: '1939870510970987',
          leftShares: '0',
        },
      ],
    },
  ],
  exchangeRate: '1.010906701603882254',
};

export const MOCK_GET_POOLED_STAKES_API_RESPONSE_HIGH_ASSETS_AMOUNT: PooledStakes = {
  accounts: [
    {
      account: '0x0111111111abcdef2222222222abcdef33333333',
      lifetimeRewards: '0',
      assets: '99999999990000000000000',
      exitRequests: [],
    },
  ],
  exchangeRate: '1.010906701603882254',
};

export const MOCK_GET_VAULT_RESPONSE: VaultData = {
  apy: '2.853065141088762750393474836309926',
  capacity:
    '12345678901234567890123456789012345678901234567890123456789012345678901234567890123456',
  feePercent: 1500,
  totalAssets: '7723070453364602130892',
  vaultAddress: '0x0a1b2c3d4e5f6a7b8c9dabecfd0123456789abcd',
};

export const MOCK_STAKING_EARNINGS_DATA = {
  ANNUAL_EARNING_RATE: '2.6%',
  LIFETIME_REWARDS: {
    FIAT: '$2',
    ETH: '0.02151 ETH',
  },
  EST_ANNUAL_EARNINGS: {
    FIAT: '$15.93',
    ETH: '0.0131 ETH',
  },
};

export const MOCK_REWARD_DATA = {
  REWARDS: {
    ETH: '0.13 ETH',
    FIAT: '$334.93',
  },
};

export const MOCK_STAKING_API_SERVICE: Partial<StakingApiService> = {
  fetchFromApi: jest.fn(),
  getPooledStakes: jest.fn(),
  getVaultData: jest.fn(),
  getPooledStakingEligibility: jest.fn(),
  baseUrl: 'https://staking.api.com',
};

const MOCK_POOLED_STAKING_CONTRACT_SERVICE = {
  chainId: ChainId.ETHEREUM,
  connectSignerOrProvider: jest.fn(),
  contract: new Contract('0x0000000000000000000000000000000000000000', []),
  convertToShares: jest.fn(),
  encodeClaimExitedAssetsTransactionData: jest.fn(),
  encodeDepositTransactionData: jest.fn(),
  encodeEnterExitQueueTransactionData: jest.fn(),
  encodeMulticallTransactionData: jest.fn(),
  estimateClaimExitedAssetsGas: jest.fn(),
  estimateDepositGas: jest.fn(),
  estimateEnterExitQueueGas: jest.fn(),
  estimateMulticallGas: jest.fn(),
};

export const MOCK_POOL_STAKING_SDK: Stake = {
  stakingContract: MOCK_POOLED_STAKING_CONTRACT_SERVICE,
  stakingApiService: MOCK_STAKING_API_SERVICE as StakingApiService,
  sdkType: StakingType.POOLED,
  setSdkType: jest.fn(),
};
