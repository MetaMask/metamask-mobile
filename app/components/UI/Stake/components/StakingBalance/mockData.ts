import { TokenI } from '../../../../UI/Tokens/types';
import {
  GetStakesApiResponse,
  GetVaultDataApiResponse,
} from './StakingBalance.types';

// TODO: Replace mock data when connecting to backend.
export const MOCK_STAKED_ETH_ASSET = {
  balance: '4.9999 ETH',
  balanceFiat: '$13,292.20',
  name: 'Staked Ethereum',
  symbol: 'ETH',
} as TokenI;

// TODO: Replace mock data when connecting to backend.
export const MOCK_GET_POOLED_STAKES_API_RESPONSE: GetStakesApiResponse = {
  accounts: [
    {
      account: '0x0123456789abcdef0123456789abcdef01234567',
      lifetimeRewards: '43927049303048',
      assets: '17913326707142320',
      exitRequests: [
        {
          // Unstaking
          positionTicket: '2153260738145148336740',
          timestamp: '1727110415000',
          totalShares: '989278156820374',
          withdrawalTimestamp: null,
          exitQueueIndex: '-1',
          claimedAssets: null,
          leftShares: null,
        },
        // Requests below are claimable.
        {
          positionTicket: '515964521392314631201',
          timestamp: '1720539311000',
          totalShares: '99473618267007',
          withdrawalTimestamp: '0',
          exitQueueIndex: '57',
          claimedAssets: '100006626507361',
          leftShares: '0',
        },
        {
          positionTicket: '515964620865932898208',
          timestamp: '1720541495000',
          totalShares: '99473618267007',
          withdrawalTimestamp: '0',
          exitQueueIndex: '57',
          claimedAssets: '100006626507361',
          leftShares: '0',
        },
        {
          positionTicket: '516604671289934191921',
          timestamp: '1720607327000',
          totalShares: '1929478758729790',
          withdrawalTimestamp: '0',
          exitQueueIndex: '58',
          claimedAssets: '1939870510970987',
          leftShares: '0',
        },
      ],
    },
  ],
  exchangeRate: '1.010906701603882254',
};

// TODO: See if this data is available yet. If not, mock backend response.
export const MOCK_GET_VAULT_RESPONSE: GetVaultDataApiResponse = {
  apy: '2.853065141088762750393474836309926',
  capacity:
    '12345678901234567890123456789012345678901234567890123456789012345678901234567890123456',
  feePercent: 1500,
  totalAssets: '7723070453364602130892',
  vaultAddress: '0x0a1b2c3d4e5f6a7b8c9dabecfd0123456789abcd',
};
