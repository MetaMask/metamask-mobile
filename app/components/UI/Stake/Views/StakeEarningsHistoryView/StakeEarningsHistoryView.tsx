import { useNavigation, useRoute } from '@react-navigation/native';
import React, { useEffect } from 'react';
import { View } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { strings } from '../../../../../../locales/i18n';
import { useStyles } from '../../../../hooks/useStyles';
import { getStakingNavbar } from '../../../Navbar';
import StakingEarningsHistory from '../../components/StakingEarnings/StakingEarningsHistory/StakingEarningsHistory';
import styleSheet from './StakeEarningsHistoryView.styles';
import { StakeEarningsHistoryViewRouteParams } from './StakeEarningsHistoryView.types';

const StakeEarningsHistoryView = () => {
  const navigation = useNavigation();
  const route = useRoute<StakeEarningsHistoryViewRouteParams>();
  const { styles, theme } = useStyles(styleSheet, {});
  const { asset } = route.params;

  useEffect(() => {
    navigation.setOptions(
      getStakingNavbar(
        strings('stake.earnings_history_title', {
          ticker: asset.ticker,
        }),
        navigation,
        theme.colors,
        {
          backgroundColor: theme.colors.background.default,
          hasCancelButton: false,
          hasBackButton: true,
        },
      ),
    );
  }, [navigation, theme.colors, asset.ticker]);

  return (
    <ScrollView contentContainerStyle={styles.mainContainer}>
      <View>
        <StakingEarningsHistory asset={asset} />
      </View>
    </ScrollView>
  );
};

export default StakeEarningsHistoryView;
