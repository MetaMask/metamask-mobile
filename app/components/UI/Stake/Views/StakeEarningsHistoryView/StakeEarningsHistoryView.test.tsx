import React from 'react';
import StakeEarningsHistoryView from './StakeEarningsHistoryView';
import useStakingEarningsHistory from '../../hooks/useStakingEarningsHistory';
import { MOCK_STAKED_ETH_ASSET } from '../../__mocks__/mockData';
import { fireLayoutEvent } from '../../../../../util/testUtils/react-native-svg-charts';
import { getStakingNavbar } from '../../../Navbar';
import renderWithProvider from '../../../../../util/test/renderWithProvider';
import { backgroundState } from '../../../../../util/test/initial-root-state';
import { Hex } from '@metamask/utils';

jest.mock('../../../Navbar');
jest.mock('../../hooks/useStakingEarningsHistory');

const mockNavigation = {
  navigate: jest.fn(),
  setOptions: jest.fn(),
};

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => mockNavigation,
    useRoute: () => ({
      key: '1',
      name: 'params',
      params: { asset: MOCK_STAKED_ETH_ASSET },
    }),
  };
});
jest.mock('react-native-svg-charts', () => {
  const reactNativeSvgCharts = jest.requireActual('react-native-svg-charts'); // Get the actual Grid component
  return {
    ...reactNativeSvgCharts,
    Grid: () => <></>,
  };
});

(useStakingEarningsHistory as jest.Mock).mockReturnValue({
  earningsHistory: [
    {
      dateStr: '2023-01-01',
      dailyRewards: '1000000000000000000',
      sumRewards: '1000000000000000000',
    },
    {
      dateStr: '2023-01-02',
      dailyRewards: '1000000000000000000',
      sumRewards: '2000000000000000000',
    },
  ],
  isLoading: false,
  error: null,
});

const mockInitialState = {
  settings: {},
  engine: {
    backgroundState: {
      ...backgroundState,
      CurrencyRateController: {
        currentCurrency: 'usd',
        currencyRates: {
          ETH: {
            conversionRate: 3363.79,
          },
        },
      },
      NetworkController: {
        selectedNetworkClientId: 'selectedNetworkClientId',
        networkConfigurationsByChainId: {
          '0x1': {
            nativeCurrency: 'ETH',
            chainId: '0x1' as Hex,
            rpcEndpoints: [
              {
                networkClientId: 'selectedNetworkClientId',
              },
            ],
            defaultRpcEndpointIndex: 0,
          },
        },
      },
    },
  },
};

const earningsHistoryView = <StakeEarningsHistoryView />;

describe('StakeEarningsHistoryView', () => {
  it('renders correctly and matches snapshot', () => {
    const renderedView = renderWithProvider(earningsHistoryView, {
      state: mockInitialState,
    });
    fireLayoutEvent(renderedView.root);
    expect(renderedView.toJSON()).toMatchSnapshot();
  });

  it('calls navigation setOptions to get staking navigation bar', () => {
    const renderedView = renderWithProvider(earningsHistoryView, {
      state: mockInitialState,
    });
    fireLayoutEvent(renderedView.root);
    expect(mockNavigation.setOptions).toHaveBeenCalled();
    expect(getStakingNavbar).toHaveBeenCalled();
  });
});
