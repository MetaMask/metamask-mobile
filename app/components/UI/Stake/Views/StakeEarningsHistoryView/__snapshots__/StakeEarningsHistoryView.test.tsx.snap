// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StakeEarningsHistoryView renders correctly and matches snapshot 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 1,
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingTop": 8,
    }
  }
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View>
    <View>
      <View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 32,
              "paddingTop": 32,
            }
          }
        >
          <View
            style={
              {
                "marginLeft": 8,
                "marginRight": 8,
                "width": 50,
              }
            }
          >
            <TouchableOpacity
              onPress={[Function]}
              onPressIn={[Function]}
              onPressOut={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 32,
                  "paddingHorizontal": 14,
                  "paddingVertical": 7,
                  "width": "100%",
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#141618",
                    "fontFamily": "EuclidCircularB-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 3,
                    "lineHeight": 22,
                    "textAlign": "center",
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#6a737d",
                      "fontFamily": "EuclidCircularB-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 22,
                    }
                  }
                >
                  7D
                </Text>
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              {
                "marginLeft": 8,
                "marginRight": 8,
                "width": 45,
              }
            }
          >
            <TouchableOpacity
              onPress={[Function]}
              onPressIn={[Function]}
              onPressOut={[Function]}
              style={
                {
                  "backgroundColor": "#f2f4f6",
                  "borderRadius": 32,
                  "paddingHorizontal": 14,
                  "paddingVertical": 7,
                  "width": "100%",
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#141618",
                    "fontFamily": "EuclidCircularB-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 3,
                    "lineHeight": 22,
                    "textAlign": "center",
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#6a737d",
                      "fontFamily": "EuclidCircularB-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 22,
                    }
                  }
                >
                  M
                </Text>
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              {
                "marginLeft": 8,
                "marginRight": 8,
                "width": 42,
              }
            }
          >
            <TouchableOpacity
              onPress={[Function]}
              onPressIn={[Function]}
              onPressOut={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 32,
                  "paddingHorizontal": 14,
                  "paddingVertical": 7,
                  "width": "100%",
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#141618",
                    "fontFamily": "EuclidCircularB-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 3,
                    "lineHeight": 22,
                    "textAlign": "center",
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#6a737d",
                      "fontFamily": "EuclidCircularB-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 22,
                    }
                  }
                >
                  Y
                </Text>
              </Text>
            </TouchableOpacity>
          </View>
        </View>
        <View
          onLayout={[Function]}
          testID="earnings-history-chart-container"
        >
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingBottom": 16,
                  "paddingTop": 16,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#1c8234",
                    "fontFamily": "EuclidCircularB-Bold",
                    "fontSize": 24,
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 32,
                  }
                }
              >
                2

                ETH
              </Text>
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#6a737d",
                    "fontFamily": "EuclidCircularB-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 22,
                  }
                }
              >
                Lifetime earnings
              </Text>
            </View>
            <View
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                {
                  "flex": 1,
                  "width": "100%",
                }
              }
              testID="earnings-history-chart"
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "height": 75,
                      "marginBottom": 4,
                      "paddingBottom": 16,
                      "paddingTop": 16,
                    },
                  ]
                }
              >
                <View
                  onLayout={[Function]}
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <RNSVGSvgView
                    bbHeight={100}
                    bbWidth={300}
                    focusable={false}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        {
                          "height": 100,
                          "width": 300,
                        },
                        {
                          "flex": 0,
                          "height": 100,
                          "width": 300,
                        },
                      ]
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M0,100L22.689075630252102,100L22.689075630252102,100L0,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-0"
                      />
                      <RNSVGPath
                        d="M25.210084033613445,100L47.89915966386555,100L47.89915966386555,100L25.210084033613445,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-1"
                      />
                      <RNSVGPath
                        d="M50.42016806722689,100L73.10924369747899,100L73.10924369747899,100L50.42016806722689,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-2"
                      />
                      <RNSVGPath
                        d="M75.63025210084034,100L98.31932773109244,100L98.31932773109244,100L75.63025210084034,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-3"
                      />
                      <RNSVGPath
                        d="M100.84033613445378,100L123.52941176470588,100L123.52941176470588,100L100.84033613445378,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-4"
                      />
                      <RNSVGPath
                        d="M126.05042016806722,100L148.73949579831933,100L148.73949579831933,100L126.05042016806722,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-5"
                      />
                      <RNSVGPath
                        d="M151.26050420168067,100L173.94957983193277,100L173.94957983193277,100L151.26050420168067,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-6"
                      />
                      <RNSVGPath
                        d="M176.47058823529412,100L199.1596638655462,100L199.1596638655462,100L176.47058823529412,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-7"
                      />
                      <RNSVGPath
                        d="M201.68067226890756,100L224.36974789915965,100L224.36974789915965,100L201.68067226890756,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-8"
                      />
                      <RNSVGPath
                        d="M226.890756302521,100L249.5798319327731,100L249.5798319327731,100L226.890756302521,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-9"
                      />
                      <RNSVGPath
                        d="M252.10084033613444,100L274.78991596638656,100L274.78991596638656,100L252.10084033613444,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-10"
                      />
                      <RNSVGPath
                        d="M277.3109243697479,1L300,1L300,100L277.3109243697479,100Z"
                        fill={
                          {
                            "brushRef": "bar-gradient",
                            "type": 1,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                        testID="earning-history-chart-bar-11"
                      />
                      <RNSVGDefs>
                        <RNSVGLinearGradient
                          gradient={
                            [
                              0,
                              1868340,
                              1,
                              438075956,
                            ]
                          }
                          gradientTransform={null}
                          gradientUnits={0}
                          name="bar-gradient"
                          x1="0%"
                          x2="0%"
                          y1="100%"
                          y2="0%"
                        />
                      </RNSVGDefs>
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-0"
                        x1={0}
                        x2={22.689075630252102}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-1"
                        x1={25.210084033613445}
                        x2={47.89915966386555}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-2"
                        x1={50.42016806722689}
                        x2={73.10924369747899}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-3"
                        x1={75.63025210084034}
                        x2={98.31932773109244}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-4"
                        x1={100.84033613445378}
                        x2={123.52941176470588}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-5"
                        x1={126.05042016806722}
                        x2={148.73949579831933}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-6"
                        x1={151.26050420168067}
                        x2={173.94957983193277}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-7"
                        x1={176.47058823529412}
                        x2={199.1596638655462}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-8"
                        x1={201.68067226890756}
                        x2={224.36974789915965}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-9"
                        x1={226.890756302521}
                        x2={249.5798319327731}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-10"
                        x1={252.10084033613444}
                        x2={274.78991596638656}
                        y1={99.5}
                        y2={99.5}
                      />
                      <RNSVGLine
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={
                          {
                            "payload": 4280058420,
                            "type": 0,
                          }
                        }
                        strokeWidth={1}
                        testID="earning-history-chart-line-11"
                        x1={277.3109243697479}
                        x2={300}
                        y1={0.5}
                        y2={0.5}
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 1,
              "paddingBottom": 35,
              "paddingTop": 24,
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "color": "#141618",
                "fontFamily": "EuclidCircularB-Bold",
                "fontSize": 14,
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 22,
              }
            }
            testID="label"
          >
            Payout history
          </Text>
          <View>
            <View
              style={
                {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#141618",
                    "fontFamily": "EuclidCircularB-Medium",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 22,
                  }
                }
                testID="label"
              >
                2023
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingTop": 10,
                    "width": "50%",
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#141618",
                      "fontFamily": "EuclidCircularB-Medium",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 22,
                    }
                  }
                  testID="label"
                >
                  January
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "justifyContent": "space-between",
                    "width": "50%",
                  }
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#1c8234",
                        "fontFamily": "EuclidCircularB-Medium",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 22,
                      }
                    }
                    testID="label"
                  >
                    +
                    2

                    ETH
                  </Text>
                </View>
                <View>
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#6a737d",
                        "fontFamily": "EuclidCircularB-Regular",
                        "fontSize": 12,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                      }
                    }
                  >
                    $6727.58
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
